### ISSUE TITLE
[BUG] Adding a template crashes 3.0.1

### ISSUE THREAD

**Issue by powerje** [Azure/azure-notificationhubs-ios#93]
**Describe the bug**

Using the 3.0.1 SDK I'm seeing a crash when adding a template.

***Exception or Stack Trace***
```
2020-08-10 16:30:58.202232-0400 MyApp [4715:2125173] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Invalid type in JSON write (MSInstallationTemplate)'
*** First throw call stack:
(0x1a72fd654 0x1a701fbcc 0x1a7775b7c 0x1a77784a0 0x1a734ac14 0x1a71d6fc4 0x1a77779d0 0x1a77784a0 0x1a734ac14 0x1a71d551c 0x1a77779d0 0x1a762d00c 0x1a762cc90 0x1006ab280 0x1006a43e8 0x1006a3708 0x1a76f6d94 0x1a727c134 0x1a727be50 0x1a727b52c 0x1a727653c 0x1a7275ba8 0x1b13e5344 0x1ab3b13e4 0x100146144 0x1a70fd8f0)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

[MSInstallation.m](https://github.com/Azure/azure-notificationhubs-ios/blob/95a407e56dbe1d89582b0312034001f462870e28/WindowsAzureMessaging/WindowsAzureMessaging/Model/MSInstallation.m#L115) is converting a dictionary to JSON but one of the elements (templates array) of the dictionary isn't a supported type, causing the crash:

```
(lldb) po dictionary
{
    expirationTime = "2020-11-08T15:33-0500";
    installationId = "some_id";
    platform = apns;
    pushChannel = device_token;
    tags =     (
        "my_tag"
    );
    templates =     {
        template = "<MSInstallationTemplate: 0x2839ab5a0>";
    };
}
```

**To Reproduce**
Run the code in the code snippet.

***Code Snippet***
```
        let body =
        """
        {
          "aps": {
            "alert": "$(message)"
          }
        }
        """
        let template = MSInstallationTemplate()
        template.body = body
        template.addTag(azureTag)
        MSNotificationHub.setTemplate(template, forKey: "template")
```

**Expected behavior**
I expect the template to register and no crash.

**Setup (please complete the following information):**
 - OS: iOS 13.6
 - IDE : Xcode 11.6
 - Version of the Library used: 3.0.1 via SPM

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by powerje** [comment]
I was just creating a PR for this myself 😅 

Thanks for jumping on it so quickly

**Comment by mpodwysocki** [comment]
Fixed as per #94
